### PR TITLE
Run tests on CI

### PR DESCRIPTION
**Pull request by igchor**
This change is [Reviewable](https://reviewable.io/reviews/pmem/cachelib/6)

### CODE REVIEW

File: .github/workflows/build-cachelib-centos.yml [hunk 1]
```diff
@@ -34,3 +34,5 @@ jobs:
         uses: actions/checkout@v2
       - name: "build CacheLib using build script"
         run: ./contrib/build.sh -j -v -T
+      - name: "run tests"
+        run: cd opt/cachelib && ../../run_tests.sh
```
Comment (victoria-mcgrath) [inline .github/workflows/build-cachelib-centos.yml:38]: Please add timeout for this step.

File: run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+BLACKLIST="allocator-test-NvmCacheTests
+	common-test-TimeTests
+	shm-test-test_page_size"
+
+find -type f \( -not -name "*bench*" \) -executable | grep -vF "$BLACKLIST" | xargs -n1 bash -c
```
Comment (victoria-mcgrath) [inline run_tests.sh:5]: Will it be possible to show some output of this command so that we can see which tests were running / passing / failing?

**Review by guptask — CHANGES_REQUESTED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/cachelib/6)* status: all files reviewed, 3 unresolved discussions (waiting on @igchor)

___
*[run_tests.sh, line 2 at r1](https://reviewable.io/reviews/pmem/cachelib/6#-MnlY7Yr3x_zgmGBYOiv:-MnlY7Yr3x_zgmGBYOiw:b7yl70z) ([raw file](https://github.com/pmem/cachelib/blob/ba09746a76d66b2a2be9e734a9fc5d6ca05dc842/run_tests.sh#L2)):*
> ```Shell
> BLACKLIST="allocator-test-NvmCacheTests
> 	common-test-TimeTests
> ```

isn't "\"
 needed after end of each line ?

**Review by igchor — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/cachelib/6)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @guptask and @victoria-mcgrath)

___
*[run_tests.sh, line 2 at r1](https://reviewable.io/reviews/pmem/cachelib/6#-MnlY7Yr3x_zgmGBYOiv:-MnytzJ2BKRWEpVHiLFh:b-rqz77l) ([raw file](https://github.com/pmem/cachelib/blob/ba09746a76d66b2a2be9e734a9fc5d6ca05dc842/run_tests.sh#L2)):*
<details><summary><i>Previously, guptask (Sounak Gupta) wrote…</i></summary><blockquote>

isn't "\"
 needed after end of each line ?
</blockquote></details>

No, the idea here is to separate each test by a newline - I added a comment.
___
*[run_tests.sh, line 5 at r1](https://reviewable.io/reviews/pmem/cachelib/6#-MnkHBhfTHZ8M6FpWRLa-r1-5:-Mnyu3830KmxudTFjGC7:b-x1jc65) ([raw file](https://github.com/pmem/cachelib/blob/ba09746a76d66b2a2be9e734a9fc5d6ca05dc842/run_tests.sh#L5)):*
<details><summary><i>Previously, victoria-mcgrath wrote…</i></summary><blockquote>

Will it be possible to show some output of this command so that we can see which tests were running / passing / failing?
</blockquote></details>

The output is already visible here: https://github.com/pmem/CacheLib/runs/4118303221?check_suite_focus=true
I don't know why this does not work on centos...
___
*[.github/workflows/build-cachelib-centos.yml, line 38 at r1](https://reviewable.io/reviews/pmem/cachelib/6#-MnkGRYRaDeBt073tZx4-r1-38:-MnyyZ4n8O0Y5vwy0nKQ:b-896fix) ([raw file](https://github.com/pmem/cachelib/blob/ba09746a76d66b2a2be9e734a9fc5d6ca05dc842/.github/workflows/build-cachelib-centos.yml#L38)):*
<details><summary><i>Previously, victoria-mcgrath wrote…</i></summary><blockquote>

Please add timeout for this step. 
</blockquote></details>

Done.

**Review by igchor — COMMENTED** [review]
I've also disabled navy tests for now to speed up the build - do you think it's OK or should we keep it? We can maybe run them periodically on master instead?

> *[Reviewable](https://reviewable.io/reviews/pmem/cachelib/6#-:-Mnz0A3vDhmBwOT2m-F8:brjcffe)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @guptask and @victoria-mcgrath)

**Review by igchor — COMMENTED** [review]
I've added this change on top of my Config API integration PR to see if it builds here: https://github.com/pmem/CacheLib/pull/8

> *[Reviewable](https://reviewable.io/reviews/pmem/cachelib/6#-:-Mnzh1in5vhvkIowyKe-:bdtnack)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @guptask and @victoria-mcgrath)

___
*[run_tests.sh, line 5 at r1](https://reviewable.io/reviews/pmem/cachelib/6#-MnkHBhfTHZ8M6FpWRLa-r1-5:-Mnzh-NP5cdAVa6WHpCi:b-hoo3cj) ([raw file](https://github.com/pmem/cachelib/blob/ba09746a76d66b2a2be9e734a9fc5d6ca05dc842/run_tests.sh#L5)):*
<details><summary><i>Previously, igchor (Igor Chorążewicz) wrote…</i></summary><blockquote>

The output is already visible here: https://github.com/pmem/CacheLib/runs/4118303221?check_suite_focus=true
I don't know why this does not work on centos...
</blockquote></details>

Ok, it works now, I just messed up the path.

**Comment by victoria-mcgrath** [comment]
Yes, let's disable all long tests and schedule periodic full test runs on develop.

**Pull request merged**